core/src/main/resources/jenkins/formelementpath/form-element-path.js

```diff
@@ -0,0 +1,229 @@
+                        if (on != null) {
+                            addProperty(p, on, e);
+                        } else {
+                            addProperty(p, e.name, e);
+                        }
+                    }
+                    // switch to multipart/form-data to support file submission
+                    // @enctype is the standard, but IE needs @encoding.
+                    form.enctype = form.encoding = "multipart/form-data";
+                    break;
+                // otherwise fall through
+                default:
+                    p = findParent(e);
+                    addProperty(p, e.name, e);
+                    break;
+            }
+        }
+
+        function annotate(e, path) {
+            e.setAttribute("path", path);
+            var o = e.formDom || {};
+            for (var key in o) {
+                var v = o[key];
+
+                function child(v, i) {
+                    var suffix = null;
+                    var newKey = key;
+                    if (v.parentNode.className && v.parentNode.className.indexOf("one-each") > -1 && v.parentNode.className.indexOf("honor-order") > -1) {
+                        suffix = v.getAttribute("descriptorId").split(".").pop()
+                    } else if (v.getAttribute("type") == "radio") {
+                        suffix = v.value
+                        while (newKey.substring(0, 8) == 'removeme')
+                            newKey = newKey.substring(newKey.indexOf('_', 8) + 1);
+                    } else if (v.getAttribute("suffix") != null) {
+                        suffix = v.getAttribute("suffix")
+                    } else {
+                        if (i > 0)
+                            suffix = i;
+                    }
+                    if (suffix == null) suffix = "";
+                    else suffix = '[' + suffix + ']';
+
+                    annotate(v, path + "/" + newKey + suffix);
+                }
+
+                if (v instanceof Array) {
+                    var i = 0;
+                    $A(v)._each(function (v) {
+                        child(v, i++)
+                    })
+                } else {
+                    child(v, 0)
+                }
+            }
+
+        }
+
+        annotate(form, "");
+
+        // clean up
+        for (i = 0; i < doms.length; i++)
+            doms[i].formDom = null;
+
+        return true;
+    }
+
+    function applyAll() {
+        $$("FORM")._each(function (e) {
+            buildFormTree(e);
+        })
+    }
+
+    applyAll();
+
+    // run this periodically to cope with DOM changes
+    window.setInterval(applyAll, 1000);
+
+    // in Jenkins 1.452 and onward, there's a callback available when DOM changes
+    if (typeof (layoutUpdateCallback) != "undefined") {
+        layoutUpdateCallback.add(applyAll)
+    }
+
+    // expose this globally so that Selenium can call it
+    window.recomputeFormElementPath = applyAll;
```

Comment:
Will this blow up somehow if the plugin is also installed?

Reply:
I plan to bump the baseline of the plugin and empty it out initially so this shouldn’t really matter

Reply:
it wouldn't blow up anyway just last one to register would win